### ISSUE TITLE
Only 0 of 1 slaves were synced

### ISSUE THREAD

**Issue by zhoujiale** [redisson/redisson#4222]
Hello, I am using redssion cluster using rLock. Lock (expireTime, TimeUnit. SECONDS);
Have a problem
```
Caused by: java.lang.IllegalStateException: Only 0 of 1 slaves were synced
	at org.redisson.RedissonBaseLock.lambda$evalWriteAsync$0(RedissonBaseLock.java:232)
	at java.util.concurrent.CompletableFuture.uniHandle(CompletableFuture.java:836)
	at java.util.concurrent.CompletableFuture$UniHandle.tryFire(CompletableFuture.java:811)
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:488)
	at java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:1975)
	at org.redisson.misc.RedissonPromise.trySuccess(RedissonPromise.java:83)
	at org.redisson.command.CommandBatchService.lambda$executeAsync$7(CommandBatchService.java:324)
	at java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:774)
	at java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:750)
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:488)
	at java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:1975)
	at org.redisson.command.RedisCommonBatchExecutor.handleResult(RedisCommonBatchExecutor.java:162)
	at org.redisson.command.RedisExecutor.checkAttemptPromise(RedisExecutor.java:476)
	at org.redisson.command.RedisExecutor.lambda$execute$3(RedisExecutor.java:172)
	at java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:774)
	at java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:750)
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:488)
	at java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:1975)
	at org.redisson.client.handler.CommandDecoder.decodeCommandBatch(CommandDecoder.java:318)
	at org.redisson.client.handler.CommandDecoder.decodeCommand(CommandDecoder.java:210)
	at org.redisson.client.handler.CommandDecoder.decode(CommandDecoder.java:137)
	at org.redisson.client.handler.CommandDecoder.decode(CommandDecoder.java:113)
	at io.netty.handler.codec.ByteToMessageDecoder.decodeRemovalReentryProtection(ByteToMessageDecoder.java:501)
	at io.netty.handler.codec.ReplayingDecoder.callDecode(ReplayingDecoder.java:366)
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:276)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:163)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:714)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:650)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:576)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:493)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	... 1 common frames omitted
```
I saw that this problem was fixed in redisson 3.17, but I still have this problem with 3.17
```
           <dependency>
                <groupId>org.redisson</groupId>
                <artifactId>redisson-spring-boot-starter</artifactId>
                <version>3.17.0</version>
            </dependency>
```
Here is my redis cluster environment
```
cluster_state:ok
cluster_slots_assigned:16384
cluster_slots_ok:16384
cluster_slots_pfail:0
cluster_slots_fail:0
cluster_known_nodes:6
cluster_size:3
cluster_current_epoch:6
cluster_my_epoch:2
cluster_stats_messages_ping_sent:5675
cluster_stats_messages_pong_sent:5600
cluster_stats_messages_sent:11275
cluster_stats_messages_ping_received:5600
cluster_stats_messages_pong_received:5671
cluster_stats_messages_received:11271
```
Can you help me find out what the problem is

**Comment by lehuuthanh5** [comment]
I think, you can use this config to ignore the exception 
`checkLockSyncedSlaves = false`

**Comment by mrniko** [comment]
https://github.com/redisson/redisson/issues/4222#issuecomment-1097600406

**Comment by zhoujiale** [comment]
[> I think, you can use this config to ignore the exception `checkLockSyncedSlaves = false`](https://github.com/redisson/redisson/issues/4222#issuecomment-1097600406)
Before I looked at the source code, also turned off this parameter to ensure that the program runs. But I wonder if there is a better solution, or what is the cause of this problem, because the official issue mentioned that this problem was solved in 3.17 before

**Comment by zhoujiale** [comment]
thanks&nbsp; for your answer


Before I looked at the source code, also turned off this parameter to ensure that the program runs. But I wonder if there is a better solution, or what is the cause of this problem, because the official issue mentioned that this problem was solved in 3.17 before




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "redisson/redisson"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年4月13日(星期三) 下午2:16
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [redisson/redisson] Only 0 of 1 slaves were synced (Issue #4222)





 
I think, you can use this config to ignore the exception
 checkLockSyncedSlaves = false

**Comment by HuiWang1995** [comment]
Hello, I also encountered this problem. So I first learned the mechanism of redis master-slave synchronization, and then read the source code of redisson. It is found that redisson waits for 1 second by default and throws an exception when the master-slave synchronization is not completed.

So, I config `options.syncTimeout` in `CommandBatchService`， which can be inject into `Redisson` .

Welcome to see my blog :
https://juejin.cn/post/7231057317332172855/
